### ISSUE TITLE
[bundle] Bundled code doesn't work as expected

### ISSUE THREAD

**Issue by MariusVatasoiu** [denoland/deno#8770]
Code bundled with `deno bundle` works different than `deno run`.

Steps to reproduce it:

This code works:
```javascript
import { resize } from 'https://deno.land/x/deno_image@v0.0.3/mod.ts';

const portraitJPG = await resize(Deno.readFileSync("./portrait.jpg"), {width: 100});
Deno.writeFileSync('./portrait.small.jpg', portraitJPG);
```

But this code doesn't:
```bash
deno bundle https://deno.land/x/deno_image@v0.0.3/mod.ts index.bundle.js
```

```javascript
import { resize } from './index.bundle.js';

const portraitJPG = await resize(Deno.readFileSync("./portrait.jpg"), {width: 100});
Deno.writeFileSync('./portrait.small.jpg', portraitJPG);
```

### Expected result:

`resize` function should return an Uint8Arrray for the resized image and than should be written on the disk.

### Actual result:

```bash
error: Uncaught Error: unknown JPEG marker f980
                        throw new Error("unknown JPEG marker " + fileMarker.toString(16));
                              ^
    at constructor.parse (file:///.../index.bundle.js:1155:31)
```

### Deno version: 1.6.1

I used this [image](https://ibb.co/SyvJTrQ) but should work with any .jpg file.

**Comment by kitsonk** [comment]
When you say "doesn't work" what happens?

What version of Deno?

**Comment by MariusVatasoiu** [comment]
@kitsonk I updated the issue with the error and Deno version.

**Comment by Nexxkinn** [comment]
Might be caused by deno messing up code execution order.

In my current version ( 1.6.0 ). This line of code in my repo:
```ts
// using oak framework
const app = new Application();
      app.use(router.routes());
      app.use(router.allowedMethods());

      app.addEventListener("listen", (ctx) => {
          console.log(`Serving requests at ${ctx.hostname}:${ctx.port}`)
      })
```

will be bundled like this

```ts
const app2 = new Application();
app2.addEventListener("listen", (ctx)=>{
    console.log(`Serving requests at ${ctx.hostname}:${ctx.port}`);
});
await app2.listen({
    hostname: config.hostname,
    port: config.port
});

// some hundreds of lines later
app2.use(router.routes());
app2.use(router.allowedMethods());

```

causing an error like this:
```
error: Uncaught (in promise) TypeError: There is no middleware to process requests.
            throw new TypeError("There is no middleware to process requests.");
                  ^
    at Application.listen (file:///C:/Users/natha/Desktop/repos/downhost/dist/downhost.js:13102:19)
    at file:///C:/Users/natha/Desktop/repos/downhost/dist/downhost.js:13161:12
```

My solution would be to not using module bundling until deno fixed this in future versions.

**Comment by FrageDev** [comment]
I've got the same problem as Nexxkinn . I'm developing an API using oak with Deno by using Typescript.
My API is working without any error (as far as I know) when I start it with `deno run --allow-all app.ts`
When I use `deno compile app.ts` and I start the app.exe I've got :

`>app
error: TypeError: There is no middleware to process requests.
    at Application.listen (file://$deno$/bundle.js:10697:19)
    at file://$deno$/bundle.js:10753:12`

**Comment by kitsonk** [comment]
cc/ @kdy1

**Comment by kdy1** [comment]
@kitsonk  I'm now working to fix this issue. (Preserving order of statements within single module would be solution to this issue.)

**Comment by kitsonk** [comment]
No problem! 😁   Hopefully the exams went well!

**Comment by kdy1** [comment]
I think this is fixed by https://github.com/swc-project/swc/pull/1264

**Comment by Nexxkinn** [comment]
Looking forward for this fix, thank you very much @kdy1 !

**Comment by kdy1** [comment]
This is fixed by new version of swc_bundler. (I've added test for it)

**Comment by FrageDev** [comment]
Hi, from what I read above this issue is supposed to be fixed. I download the latest version or Deno (1.6.3) and I tried to compile my app again.

Now the "There is no middleware to process requests" is gone, seems to be fixed. But now I've got this error. I don't know if it"s still related.

"
error: ReferenceError: DenoStdInternalError is not defined
    at assert2 (file://$deno$/bundle.js:7476:9)
    at getLogger (file://$deno$/bundle.js:7601:9)
    at file://$deno$/bundle.js:7614:16
"

App is working fine from deno console. the getLogger function is a function from https://deno.land/std@0.83.0/log
